lib/model/list_model.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:isar/isar.dart';
+import 'package:lists/model/database_manager.dart';
+import 'package:lists/model/item.dart';
+
+part 'list_model.g.dart';
+
+/// ListModel:
+///   - this class models a list in the app, such as a
+///     to-do list, a shopping list, etc.
+@Collection()
+class ListModel {
+  Id id = Isar.autoIncrement;
+  late String title;
+
+  final items = IsarLinks<Item>();
+
+  Iterable<Item> itemsView() => items;
+
+  // a zero-arg constructor is required for classes that are isar collections
+  ListModel();
+  ListModel.fromTitle(this.title);
+
+  void init() => items.loadSync();
+
+  Future<void> addOrUpdate(Item newItem) async {
```

Comment:
Can you split this into two separate `add` and `update` methods? I think it would make more sense.

Reply:
That makes sense. Do you think the `update` method should check to see if the `Item` to be updated is in `this.items` before it updates it? Or should it be assumed (a precondition)?

Reply:
That could be good. If the check failed, you should throw some kind of error so that the problem shows up to developers.